src/frontend/frontend.jl

```diff
@@ -896,27 +895,21 @@ function add_cost_gen!(
             warn(LOGGER, w * "Using id = $element_id instead.")
         end
     end
-    if ismissing(gen_id)
-        push!(net.cost_gen, Any[coeffs, element_id, model, ncost])
+    push!(net.cost_gen, Any[coeffs, element_id, model, ncost])
```

Comment:
This push should only happen if `gen_id in gen_ids`, so this line should be moved just before line 901.

Reply:
good catch! I made the changes in the new commit.